lib/api.js

```diff
@@ -1459,6 +1460,44 @@ function insertLogObject (o = {}) {
   return o
 }
 
+/**
+ * Return text delimited representation of the trace containing trace_id, span_id, trace_flags. The primary intended use for this is
+ * to custom tokens in log packages.
+ *
+ * @method ao.getLogString
+ * @param {string} [delimiter] - the delimiter to use
+ *
+ * @returns {string} - the trace log string (e.g. trace_id:... span_id: ..., trace_flages: ...)
+ *
+ * @example
+ * log4js.configure({
+ *   appenders: {
+ *     out: {
+ *       type: 'stdout',
+ *       layout: {
+ *         type: 'pattern',
+ *         pattern: '%d %p %c %x{user} says: %m is: %x{trace} %n',
+ *         tokens: {
+ *           user: function (logEvent) {
+ *             return 'Jake'
+ *           },
+ *           trace: function () {
+ *             return typeof ao !=='undefined' ? ao.getLogString() : ''
+ *           }
+ *         }
+ *       }
+ *     }
+ *   },
+ *   categories: { default: { appenders: ['out'], level: 'info' } }
+ * })
+ * const logger = log4js.getLogger()
+ * logger.info('token from api')
+ */
+function getLogString (delimiter = ' ') {
```

Comment:
should delimiter be checked so we don't expose a log injection vulnerability?

Reply:
Interesting... didn't think about that... but since it is directly passed to a native `'.join()` which in turn casts whatever it gets into a string (or if it doesn't, than this is not "our vulnerability" ), I'll leave it as is.